lib/docker-compose/models/compose_container.rb

```diff
@@ -164,6 +166,14 @@ def prepare_environment(env_entries)
   end
 
   #
+  # Forces the labels structure to use the hash format.
+  #
+  def prepare_labels(labels)
```

Comment:
Great. I currently have hakish (dirty) working setup. I'll see if I can clean it up. But I might not have the time to do that in the coming days/weeks.